### PR TITLE
Upgrade testcontainers version

### PR DESCRIPTION
**Pull request by iNikem**
This is pre-requirement for Testcontainers Cloud. Verified that it works "on my machine". As a follow up PR I will try to enable it on some of our GHA builds

### CODE REVIEW

File: instrumentation/lettuce/lettuce-5.1/testing/src/main/groovy/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceSyncClientTest.groovy [hunk 1]
```diff
@@ -90,20 +84,20 @@ abstract class AbstractLettuceSyncClientTest extends InstrumentationSpecificatio
   }
 
   def "connect"() {
-    setup:
-    RedisClient testConnectionClient = createClient(embeddedDbUri)
-    testConnectionClient.setOptions(LettuceTestUtil.CLIENT_OPTIONS)
+//    setup:
+//    RedisClient testConnectionClient = createClient(embeddedDbUri)
+//    testConnectionClient.setOptions(LettuceTestUtil.CLIENT_OPTIONS)
```
Comment (trask) [inline instrumentation/lettuce/lettuce-5.1/testing/src/main/groovy/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceSyncClientTest.groovy:89]: delete?

File: instrumentation/lettuce/lettuce-5.1/testing/src/main/groovy/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceSyncClientTest.groovy [hunk 2]
```diff
@@ -90,20 +84,20 @@ abstract class AbstractLettuceSyncClientTest extends InstrumentationSpecificatio
...
+    StatefulConnection connection = redisClient.connect()
 
     then:
     // Lettuce tracing does not trace connect
     assertTraces(0) {}
 
     cleanup:
     connection.close()
-    testConnectionClient.shutdown()
+//    testConnectionClient.shutdown()
```
Comment (trask) [inline instrumentation/lettuce/lettuce-5.1/testing/src/main/groovy/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceSyncClientTest.groovy:100]: same question

**Review by trask — APPROVED** [review]
👍

File: instrumentation/lettuce/lettuce-5.1/testing/src/main/groovy/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceSyncClientTest.groovy (hunk 1 above)
Comment (iNikem) [reply instrumentation/lettuce/lettuce-5.1/testing/src/main/groovy/io/opentelemetry/instrumentation/lettuce/v5_1/AbstractLettuceSyncClientTest.groovy]: oops, fixed

**Review by laurit — APPROVED** [review]
There is also a testcontainers dependency in benchmark-e2e/build.gradle.kts, benchmark-overhead/build.gradle.kts, examples/distro/smoke-tests/build.gradle and examples/extension/build.gradle

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by trask** [comment]
> There is also a testcontainers dependency in benchmark-e2e/build.gradle.kts, benchmark-overhead/build.gradle.kts, examples/distro/smoke-tests/build.gradle and examples/extension/build.gradle

oops, I hit merge too soon, opened #4737 to track this

**Comment by iNikem** [comment]
> > There is also a testcontainers dependency in benchmark-e2e/build.gradle.kts, benchmark-overhead/build.gradle.kts, examples/distro/smoke-tests/build.gradle and examples/extension/build.gradle
> 
> oops, I hit merge too soon, opened #4737 to track this

Already fixed in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/4727/commits/7df326c57293dfceabbc68f1421cf36ecd7daff3

**Pull request merged**